clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1030,6 +1030,7 @@ public void assign(Collection<TopicPartition> partitions) {
      * @return The fetched records (may be empty)
      */
     private Map<TopicPartition, List<ConsumerRecord<K, V>>> pollOnce(long timeout) {
+        client.maybeTriggerWakeup();
```

Comment:
@ijuma right ..., I think now I understand @hachikuji 's suggestion better :)

I guess to be able to test `!records.isEmpty()` we would have to add a `hasFetchedRecords` method to the fetcher (to avoid `poll`ing and subsequently throwing?).  Wouldn't I have to do similar locking in the Fetcher for that, though ? (I really don't have a good understanding of what locking I'd need there -> only reason I did it this way)